src/lib/kube_job.ml

```diff
@@ -85,47 +98,111 @@ let start ~log t =
       "cpu", `String (Int.to_string spec.cpus);
     ] in
   let json : Yojson.Safe.json =
-    `Assoc [
-      "kind", `String "Pod";
-      "apiVersion", `String "v1";
-      "metadata", `Assoc [
-        "name", `String spec.id;
-        "labels", `Assoc [
-          "app", `String spec.id;
+    let secrets =
+      List.filter_map spec.volume_mounts ~f:(function
+        | `Nfs _ -> None
+        | `Constant f ->
+          Some  (`Assoc [
+              "apiVersion", `String "v1";
+              "kind", `String "Secret";
+              "metadata", `Assoc [
+                "name", `String (File_contents_mount.id f);
+              ];
+              "data", `Assoc [
+                Filename.basename (File_contents_mount.path f),
+                `String (File_contents_mount.contents f |> B64.encode);
+              ];
+            ] )) in
+    let items =
```

Comment:
This whole JSON blob is getting huge and crazy; any way to sanely split it up and maybe document it a bit?